src/common/transformations/src/transformations/common_optimizations/optimize_strided_slice.cpp

```diff
@@ -274,30 +274,47 @@ struct SliceWithAttrs {
 };
 
 bool slice_is_suitable_for_optimization(const std::shared_ptr<ov::op::v8::Slice>& op, SliceAttrs& attrs) {
-    const auto& data_rank = op->get_input_partial_shape(0).rank();
+    const auto& input_shape = op->get_input_partial_shape(0);
+    const auto& data_rank = input_shape.rank();
     if (op->get_input_size() != 5 || data_rank.is_dynamic())
         return false;
+    const auto rank = data_rank.get_length();
 
-    for (size_t i = 1; i < 5; ++i) {
-        auto input_as_constant = ov::as_type_ptr<ov::op::v0::Constant>(op->get_input_node_shared_ptr(i));
-        if (!input_as_constant)
+    auto get_scalar = [](const std::shared_ptr<ov::Node>& node, int64_t& value) -> bool {
+        auto constant = ov::as_type_ptr<ov::op::v0::Constant>(node);
+        if (!constant)
             return false;
-        if (shape_size(input_as_constant->get_shape()) != 1)
+        if (shape_size(constant->get_shape()) != 1)
             return false;
+        value = constant->cast_vector<int64_t>()[0];
+        return true;
+    };
+
+    enum { START = 1, STOP, STRIDE, AXIS };
+
+    int64_t stride;
+    if (!get_scalar(op->get_input_node_shared_ptr(STRIDE), stride) || stride != 1)
+        return false;
+    if (!get_scalar(op->get_input_node_shared_ptr(AXIS), attrs.axis))
+        return false;
+    attrs.axis = attrs.axis >= 0 ? attrs.axis : attrs.axis + rank;
 
-        int64_t value = input_as_constant->cast_vector<int64_t>()[0];
+    if (input_shape[attrs.axis].is_dynamic())
+        return false;
+    const auto dimension = input_shape[attrs.axis].get_length();
 
-        if (((i == 1 || i == 2) && value < 0) || (i == 3 && value != 1))
+    for (int i = START; i <= STOP; i++) {
+        int64_t value;
+        if (!get_scalar(op->get_input_node_shared_ptr(i), value))
             return false;
-        else if (i == 1)
+        value = value >= 0 ? value : value + dimension;
+        value = std::min(value, dimension);
```

Comment:
In case my dimension is 10 and start is -100 what would happen here? I think we still need to have the check for value being negative.

Reply:
Good point! it looks like we need to clamp it first
```
value = max(min(value, dimension), -dimension); // maybe max(..., -dimension +1) ?
value = value >= 0 ? value : value + dimension;
```
@mateusztabaka could you check this?

Reply:
yes, we should clamp the value